ui/src/importer/importerTaskDetails/EitemImportDetails.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import _isEmpty from 'lodash/isEmpty';
+import { Link } from 'react-router-dom';
+import { Button, Header, List, Modal } from 'semantic-ui-react';
+import { BackOfficeRoutes } from '@inveniosoftware/react-invenio-app-ils';
+
+export class EitemImportDetails extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { open: false };
+  }
+  render() {
+    const { eitemReport } = this.props;
+    const { open } = this.state;
+    return (
+      <Modal
```

Comment:
Not a fan of repeating modals inside for loops, these are extra components that are rendered, attached to the DOM, and it could be avoided.
Modals should be one unique only at the page level, that we fill, open and close as needed.

Reply:
I tried to find where is the for loop generating those modals, do you mean that we have multiple modals displaying JSON code, rendered in the table ?